src/nvim/extmark.c

```diff
@@ -132,14 +132,15 @@ void extmark_set(buf_T *buf, uint32_t ns_id, uint32_t *idp, int row, colnr_T col
   } else if (decor) {
     mark.hl_id = decor->hl_id;
     // workaround: see above
-    mark.flags = (uint16_t)(mark.flags | (decor->hl_eol ? (uint16_t)MT_FLAG_HL_EOL : (uint16_t)0));
+    mark.flags = (uint16_t)(mark.flags | (decor->hl_eol ? (uint16_t)MT_FLAG_HL_EOL : (uint16_t)0)
+                            | (no_undo ? (uint16_t)MT_FLAG_NO_UNDO : 0));
     mark.priority = decor->priority;
   }
 
   marktree_put(buf->b_marktree, mark, end_row, end_col, end_right_gravity);
 
 revised:
-  if (op != kExtmarkNoUndo) {
+  if (op != kExtmarkNoUndo && !no_undo) {
```

Comment:
As far as I could tell (while testing out a similar patch), I noticed that `op` is never not equal to `kExtmarkNoUndo`. Is this block no longer necessary or for future use?

Reply:
good catch, away it goes.